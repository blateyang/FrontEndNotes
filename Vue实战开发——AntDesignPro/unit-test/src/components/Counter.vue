<template>
  <div>
    <span>count: {{count}}</span>
    <button @click="count++">count++</button>
  </div>
</template>
<script>
export default {
  data() {
    return {
      count: 0
    }
  },
  components: {

  }
}
</script>

<style>

</style>