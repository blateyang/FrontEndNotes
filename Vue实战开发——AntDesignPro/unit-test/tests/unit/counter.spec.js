import { mount } from "@vue/test-utils";
import Counter from "@/components/Counter.vue";

describe("Counter.vue", () => {
  const wrapper = mount(Counter);
  it("render counter html", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("count++", ()=> {
    const button = wrapper.find("button");
    button.trigger("click");
    expect(wrapper.vm.count).toBe(1);
  });
});
