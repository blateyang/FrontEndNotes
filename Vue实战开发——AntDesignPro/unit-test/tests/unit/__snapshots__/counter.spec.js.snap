// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter.vue generate counter snapshot 1`] = `<div><span>count: 0</span> <button>count++</button></div>`;

exports[`Counter.vue render counter html 1`] = `<div><span>count: 0</span> <button>count++</button></div>`;
